### DISCUSSION TITLE
Filament level

### DISCUSSION THREAD

**Discussion by clumsyzombie** [greghesp/ha-bambulab#627] [Ideas]
Question when you are in Bambu Lab Studio and are at the Device Screen and you are using Bambu Filament you can see the amount with a small line of what is left. If there any senser or any way tio pull this into Home assist to display the left of Filament left on a spool with out a cam on the AMS? I am not sure what I am even asking about but I was just thinking if that is something that can be pulled from some where. 

Or is there away to have a Filament inventory systems that monitors what Filament you have even in the closet and when you start to get low you get an alert to reorder blah blah Filament

**Comment by t3chguy** [comment]
Yes, that info is already present in this integration. 

![image](https://github.com/user-attachments/assets/b67ec307-e51a-45e5-9c03-7b17699dfa34)

**Reply by clumsyzombie** [reply]
So can you make a senser to grab like X1C-ClumsysLab-1-AMS Tray 3 and only grab Remain filament % 
So I guess can you have a senser grab set attributes under a senser 
So you can truly track the % left?

**Reply by clumsyzombie** [reply]
I got it figuered out sorry thanks a ton 

sensor:
  - platform: template
    sensors:
      spool_remaining_filament_tray_1:
        friendly_name: "Spool Remaining Filament Tray 1"
        unit_of_measurement: '%'
        value_template: "{{ state_attr('sensor.X1C-ClumsysLab-1-AMS_Tray_1', 'Remaining filament') }}"
      spool_remaining_filament_tray_2:
        friendly_name: "Spool Remaining Filament Tray 2"
        unit_of_measurement: '%'
        value_template: "{{ state_attr('sensor.X1C-ClumsysLab-1-AMS_Tray_2', 'Remaining filament') }}"
      spool_remaining_filament_tray_3:
        friendly_name: "Spool Remaining Filament Tray 3"
        unit_of_measurement: '%'
        value_template: "{{ state_attr('sensor.X1C-ClumsysLab-1-AMS_Tray_3', 'Remaining filament') }}"
      spool_remaining_filament_tray_4:
        friendly_name: "Spool Remaining Filament Tray 4"
        unit_of_measurement: '%'
        value_template:

**Comment by clumsyzombie** [comment]
Well I thought that code would work and it will not pull it so i am back tot he drawing board. 

sensor:
  - platform: template
    sensors:
      spool_remaining_filament_tray_1:
        friendly_name: "Spool Remaining Filament Tray 1"
        unit_of_measurement: '%'
        value_template: "{{ state_attr('sensor.X1C-ClumsysLab-1-AMS_Tray_1', 'Remaining filament') | float }}"
      spool_remaining_filament_tray_2:
        friendly_name: "Spool Remaining Filament Tray 2"
        unit_of_measurement: '%'
        value_template: "{{ state_attr('sensor.X1C-ClumsysLab-1-AMS_Tray_2', 'Remaining filament') | float }}"
      spool_remaining_filament_tray_3:
        friendly_name: "Spool Remaining Filament Tray 3"
        unit_of_measurement: '%'
        value_template: "{{ state_attr('sensor.X1C-ClumsysLab-1-AMS_Tray_3', 'Remaining filament') | float }}"
      spool_remaining_filament_tray_4:
        friendly_name: "Spool Remaining Filament Tray 4"
        unit_of_measurement: '%'
        value_template: "{{ state_attr('sensor.X1C-ClumsysLab-1-AMS_Tray_4', 'Remaining filament') | float }}"

**Comment by clumsyzombie** [comment]
I got it working thank you
